src/d/actor/d_a_player_main.cpp

```diff
@@ -2227,17 +3988,96 @@ BOOL daPy_lk_c::changeFrontWallTypeProc() {
 
 /* 8010FEC4-8010FFB0       .text changeSlideProc__9daPy_lk_cFv */
 int daPy_lk_c::changeSlideProc() {
-    /* Nonmatching */
+    short sVar4;
+
+    cM3dGPla* pfVar1 = getSlidePolygon();
+    if (pfVar1 != NULL) {
+        sVar4 = cM_atan2s(pfVar1->GetNP()->x, pfVar1->GetNP()->z);
+        if (m34C3 != 0) {
+            cLib_addCalc(&mVelocity, 0.0f, 0.4f, 5.0f, 1.0f);
+            m3526--;
+            if ((mVelocity > 5.0f) || (m3526 > 0)) {
+                return false;
+            }
+        }
+        if (cLib_distanceAngleS(sVar4, shape_angle.y) < daPy_HIO_slide_c0::m.field_0x8) {
+            return procSlideFront_init(sVar4);
+        } else {
+            return procSlideBack_init(sVar4);
+        }
+    } else {
+        m3526 = 8;
+        return false;
+    }
 }
 
 /* 8010FFB0-80110028       .text changeWaitProc__9daPy_lk_cFv */
-void daPy_lk_c::changeWaitProc() {
-    /* Nonmatching */
+BOOL daPy_lk_c::changeWaitProc() {
+    if (checkGrabAnime()) {
+        procGrabWait_init();
+    } else if (checkUpperAnime(LKANM_BCK_BOOMCATCH)) {
+        procBoomerangCatch_init();
+    } else if (daPy_getPlayerActorClass() != this) {
+        procControllWait_init();
+    } else {
+        procWait_init();
+    }
+    return;
```

Comment:
All of these branches should return the return value of the called function since this isn't a void function. And this void `return;` at the end isn't needed.